### PR TITLE
Quotes added for spaces in directory names.

### PR DESCRIPTION
**Pull request by nathanmaldrich**

### CODE REVIEW

**Comment by AlDanial** [comment]
I'm not understanding what exactly the quotes around the substitute operation solve, can you give an example?

**Comment by nathanmaldrich** [comment]
When doing a comparison against a directory with spaces in the name I was seeing errors about file not found on "Files/Setup.dbg". The file was really in "../Script Files/Setup.dbg". The quotes allowed for spaces to be substituted correctly.

**Comment by AlDanial** [comment]
The substitution in question should trim trailing whitespace. I'm not convinced that adding single quotes does something desireable.  Using this test program it appears the single quotes suppress the space trimming entirely:
<pre>
#!/usr/bin/perl
use warnings;
use strict;

my $a = "abc  ";
my $b = "ab cd  ";

print "a before=[$a]  "; $a =~ s/\s+$//; print "a after=[$a]\n";
print "b before=[$b]  "; $b =~ s/\s+$//; print "b after=[$b]\n";

$a = "abc  ";
$b = "ab cd  ";

print "a before=[$a]  "; $a =~ 's/\s+$//'; print "a after=[$a]\n";
print "b before=[$b]  "; $b =~ 's/\s+$//'; print "b after=[$b]\n";
</pre>

The output I see is
<pre>
a before=[abc  ]  a after=[abc]
b before=[ab cd  ]  b after=[ab cd]
a before=[abc  ]  a after=[abc  ]
b before=[ab cd  ]  b after=[ab cd  ]
</pre>

In other words, the second set of outputs for a and b show no change; trailing spaces are not trimmed.

What am I overlooking?

**Comment by nathanmaldrich** [comment]
I updated the pull request, although adding the quotes the quotes fixed my problem - the underlying problem was on the git archive command. It did not having spaces in directory names. This new line will remove trailing spaces as well as escape all spaces so it can be read correctly by the git archive command.

**Comment by AlDanial** [comment]
Is the git repo you're dealing with publicly available?  I want to better understand, and trace through the Perl code, what is going on with the spaces in directory names and how these cause problems.  In the meantime I'll try making my own git repo with such directory names and see what happens.

**Comment by AlDanial** [comment]
I created a test git repo with a directory having an embedded space in the name.  Checked in some source code, and cloc was able to count it fine given a git commit hash.  If you can't point me to a repo, can you send me a list of git commands to make a test repo that demonstrates the problem?
